snap/local/register-device.sh

```diff
@@ -13,27 +13,30 @@ fi
 
 logger -t ${SNAP_NAME} "Using configuration file: ${CONFIGURATION_FILE_PATH}."
 
+# Retrieve the directory at which the configuration is stored
+CONFIGURATION_DIR_PATH="$(dirname "${CONFIGURATION_FILE_PATH}")"
```

Comment:
Quotes in quotes seems fishy. Should the inner quotes be escaped?

Reply:
Escaping them generates weird paths (e.g. "/configuration), dirname seems to handle this fine.
Adding the quotes makes sure that if the path has spaces or strange character it is still treated as a single string and does not cause weird behaviour.